mainscripts/Trainer.py

```diff
@@ -246,6 +253,14 @@ def main(args, device_args):
                                 previews.remove(preview)
                                 previews.append ( (preview_name, cv2.resize(preview_rgb, (max_w, max_h))) )
                         selected_preview = selected_preview % len(previews)
+                        #Gan-er comment:
+                        #Auto preview implementation for the update preview obsessed users. 
+                        #G---START
+                        if autopreview == True:
+                            is_waiting_preview = True
```

Comment:
I'm not sure if I really agree with adding this feature. Turning this on will cause random slowdowns that are completely unnecessary. I feel like this adds further complications to your goal of refactoring.

Reply:
Nah, it is a trivial thing adding no complication. I put it there for myself because sometimes when you do stuff like warm restarts you want to be able to watch the progress and make a judgement for optimal learning rate. It is annoying to have to press p non stop.